src/Compilers/CSharp/Portable/Symbols/TupleFieldSymbol.cs

```diff
@@ -7,87 +7,67 @@
+        protected readonly TupleTypeSymbol _containingTuple;
 
         /// <summary>
-        /// Missing underlying field is handled for error recovery
-        /// A tuple without backing fields is usable for binding purposes, since we know its name and type,
-        /// but caller is supposed to report some kind of error at declaration.
-        ///
-        /// position is 1 for Item1.
+        /// If this field represents a tuple element (including the name match), 
+        /// id is an index of the element (zero-based).
+        /// Otherwise, (-1 - [index in members array]);
         /// </summary>
-        private TupleFieldSymbol(string name, TupleTypeSymbol containingTuple, TypeSymbol type, int position, FieldSymbol underlyingFieldOpt)
-        {
-            _name = name;
-            _containingTuple = containingTuple;
-            _underlyingFieldOpt = underlyingFieldOpt;
-            _type = type;
-            _position = position;
-        }
+        private readonly int _tupleFieldId;
 
-        /// <summary>
-        /// Copy this tuple field, but modify it to use the new containing tuple, link type and field type.
-        /// </summary>
-        internal TupleFieldSymbol WithType(TupleTypeSymbol newContainingTuple, NamedTypeSymbol newlinkType, TypeSymbol newFieldType)
+        public TupleFieldSymbol(TupleTypeSymbol container, FieldSymbol underlyingField, int tupleFieldId)
+            : base(underlyingField)
         {
-            FieldSymbol newUnderlyingFieldOpt = _underlyingFieldOpt?.OriginalDefinition.AsMember(newlinkType);
-            Debug.Assert(newUnderlyingFieldOpt == null || newUnderlyingFieldOpt.Type == newFieldType);
-
-            return new TupleFieldSymbol(_name, newContainingTuple, newFieldType, _position, newUnderlyingFieldOpt);
+            _containingTuple = container;
+            _tupleFieldId = tupleFieldId;
         }
 
-        /// <summary>
-        /// Copy this tuple field, but modify it to use the new containing tuple, and field name.
-        /// </summary>
-        internal TupleFieldSymbol WithName(TupleTypeSymbol newContainingTuple, string newName)
+        public override bool IsTupleField
         {
-            return new TupleFieldSymbol(newName, newContainingTuple, _type, _position, _underlyingFieldOpt);
+            get
+            {
+                return true;
+            }
         }
 
         /// <summary>
-        /// Helps construct a TupleFieldSymbol.
-        ///
-        /// Note that errors related to underlying field are ignored.
+        /// If this field represents a tuple element (including the name match), 
+        /// id is an index of the element (zero-based).
+        /// Otherwise, (-1 - [index in members array]);
         /// </summary>
-        internal static TupleFieldSymbol Create(string elementName, TupleTypeSymbol containingTuple, NamedTypeSymbol linkType, int fieldIndex, AssemblySymbol accessWithin)
+        public int TupleFieldId
```

Comment:
It could be useful for building the set of fields that correspond elements, for example. I decided to expose the data inside compiler since we have it.

---
In reply to: [61793423](https://github.com/dotnet/roslyn/pull/11006#discussion_r61793423) [](ancestors = 61793423)